### PR TITLE
Refactor backend

### PR DESCRIPTION
**Pull request by glaslos**
Moved the backend into a separate package. There is still a `server.go` to create and bundle the app.

### CODE REVIEW

File: backend/handlers.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+	w.WriteHeader(http.StatusAccepted)
+}
+
+type response struct {
+	User  entities.User `json:"user,omitempty"`
+	Token string        `json:"token,omitempty"`
+}
+
+// sessionHandler ...
```
Comment (dkumiszhan) [inline backend/handlers.go:47]: can you expand the comment?

**Comment by dkumiszhan** [comment]
Should we rename `src` folder to for example `frontend`?

File: public/index.html [hunk 2]
```diff
@@ -6,10 +6,10 @@
 
 	<title>Ochi</title>
 
-	<link rel='stylesheet' href='/global.css'>
-	<link rel='stylesheet' href='/build/bundle.css'>
+	<link rel='stylesheet' href='/src/global.css'>
+	<link rel='stylesheet' href='/src/build/bundle.css'>
 
-	<script defer src='/build/bundle.js'></script>
+	<script defer src='/src/build/bundle.js'></script>
```
Comment (dkumiszhan) [inline public/index.html:12]: These paths no longer work. Did you want to move them as well?

Comment (glaslos) [reply public/index.html:12]: They should still be working :thinking:

Comment (dkumiszhan) [reply public/index.html:12]: When I am running `npm run dev`, it is throwing a bunch of 404 errors. `src` directory referenced in above 4 lines no longer exists.

Comment (glaslos) [reply public/index.html]: ah, right, I was only running it through the backend! Should be fixed now.

File: server.go [hunk 3]
```diff
@@ -8,15 +8,19 @@ import (
 	"github.com/honeynet/ochi/backend"
 )
 
-//go:embed public
+//go:embed public/*
```
Comment (dkumiszhan) [inline server.go:11]: Was this done by accident?

**Comment by dkumiszhan** [comment]
LGTM

File: server.go (hunk 3 above)
Comment (glaslos) [reply server.go:11]: I was trying to get the folder content. It has the same effect and I later travers the folder structure.

**Pull request merged**